src/runner/test_case_system.py

```diff
@@ -10,11 +11,18 @@
 class TestCaseSystem:
     def run(self):
         # creates necessary objects
-        if len(sys.argv) > 2:
-            # Read YAML file
-            file_path = sys.argv[1]
-        else:
-            file_path = 'src/config.yaml'
+        parser = argparse.ArgumentParser("config file, run mode")
+        parser.add_argument("--c", type=str,
+        help="An optional argument. The configuration file")
+        parser.add_argument("--t", type=str,
+        help="An optional argument. Specifies which tests to run. By default, runs NoSIM")
```

Comment:
Thanks for adding the helps! this is hella lit

Additionally though can you specify no default test case name, and instead throw an error later if the test case is None?

I think its important that people always know what test case they are running.